Signal/src/network/PushManager.m

```diff
@@ -108,11 +109,15 @@ - (void)application:(UIApplication *)application didReceiveRemoteNotification:(N
 {
     DDLogInfo(@"%@ received remote notification", self.logTag);
 
-    [self.messageFetcherJob run];
+    [AppReadiness runNowOrWhenAppIsReady:^{
```

Comment:
Hmm, let's discuss this in person.  Feels like a tough call.  It might be just a subjective matter of taste.

Some of the benefits to the current arrangement:

* Gather concerns, e.g. push-related logic here in this push manager.
* DRY, i.e. if we pull this logic into AppDelegate we also have to pull it into ShareViewController.

Going to make a card.